### ISSUE TITLE
Real Time Phoneme Recognition

### ISSUE THREAD

**Issue by seankerman** [syl22-00/pocketsphinx.js#109]
Does pocketsphinx.js support real time phoneme recognition?

**Comment by syl22-00** [comment]
Yes, pocketsphinx (hence pocketsphinx.js) has an allophone mode, it can be triggered with `-allphone` at init time.

**Comment by JohnAReid** [comment]
Would it be possible to include an example using Phoneme ?

**Comment by syl22-00** [comment]
I think the page you point to says it all. Just (lazy) load the file (`pocketsphinx/model/en-us/en-us-phone.lm.bin`) as explained in pocketsphinx.js' README and set the init parameter (`-allphone en-us-phone.lm.bin`).

**Comment by JohnAReid** [comment]
> I think the page you point to says it all. Just (lazy) load the file

Thanks I will try it out - one question -  do I still need to load the dictionary file if I never want to use word matching?

**Comment by JohnAReid** [comment]
Downloaded the French version from 
https://sourceforge.net/projects/cmusphinx/files/Acoustic%20and%20Language%20Models/French/

and renamed the file from fr-phone.lm.dmp to phone.lm.dmp

Lazy loaded:
 

> ["/cmusphinx-fr-ptm-5.2", "phonetic", "../cmusphinx-fr-ptm-5.2/phone.lm.bin"]

seems OK no error reported.
added the following line to initialize command

> ["-allphone","phone.lm.bin"] 

reported following errors:

> INFO: ngram_model_trie.c(354): Trying to read LM in trie binary format
> ERROR: "ngram_model_trie.c", line 356: File phone.lm.bin not found

Tried different paths but not getting past the error!

**Comment by syl22-00** [comment]
You must be consistent with the file names: if you put your virtual file in a `cmusphinx-fr-ptm-5.2` folder and call it `phonetic`, you must call init with the correct file name: `["-allphone","cmusphinx-fr-ptm-5.2/phonetic"]`.

**Comment by JohnAReid** [comment]
@syl22-00 Thanks - that's a big help (well at least one more step forward)

I removed the file extension from the file and changed the lazy loading to:

> ["/cmusphinx-fr-ptm-5.2", "phonetic", "../cmusphinx-fr-ptm-5.2/phone"]

and the init call with 

> ["-allphone","cmusphinx-fr-ptm-5.2/phonetic"]

At least now it is finding the file - but the console log says:

> INFO: ngram_model_trie.c(354): Trying to read LM in trie binary format
INFO: ngram_model_trie.c(365): Header doesn't match
INFO: ngram_model_trie.c(177): Trying to read LM in arpa format
INFO: ngram_model_trie.c(70): No \data\ mark in LM file
INFO: ngram_model_trie.c(445): Trying to read LM in dmp format
INFO: ngram_model_trie.c(527): ngrams 1=38, 2=1240, 3=23231
INFO: pocketsphinx.js:1 INFO: lm_trie.c(474): Training quantizer
INFO: pocketsphinx.js:1 INFO: lm_trie.c(482): Building LM trie
INFO: pocketsphinx.js:1 INFO: allphone_search.c(236): Building PHMM net of 36 phones

It is generating output on test runs - but I think the above is trying to tell me it is not using the file.

**Comment by syl22-00** [comment]
Looks fine to me, the recognizer just tries to open your file assuming different formats until it gets the right one.

**Comment by JohnAReid** [comment]
@syl22-00 Many thanks - great to get it working - this is what I got when I tried to say the word ABRICOT

> hyp "SIL aa rr ai gg oo SIL"

I assume the SIL refers to start and finish - but no matter how hard I tried the b character to came back as bb.

**Comment by syl22-00** [comment]
That's why language models are key to speech recognition.

**Comment by syl22-00** [comment]
SIL is the "silence" phoneme.

**Comment by JohnAReid** [comment]
> SIL is the "silence" phoneme.

As Homer Simspson would say -- D'oh!
